ui/imports/Utils.qml

```diff
@@ -130,6 +130,35 @@ QtObject {
         return (hours < 10 ? "0" + hours : hours) + ":" + (minutes < 10 ? "0" + minutes : minutes)
     }
 
+    function formatDateTime(timestamp) {
+        let now = new Date()
+        let yesterday = new Date()
+        yesterday.setDate(now.getDate()-1)
+        let messageDate = new Date(Math.floor(timestamp))
+        let lastWeek = new Date()
+        lastWeek.setDate(now.getDate()-7)
+
+        let minutes = messageDate.getMinutes();
+        let hours = messageDate.getHours();
+
+        if (now.toDateString() === messageDate.toDateString()) {
+            return (hours < 10 ? "0" + hours : hours) + ":" + (minutes < 10 ? "0" + minutes : minutes)
+        } else if (yesterday.toDateString() === messageDate.toDateString()) {
+            return qsTr("Yesterday")
+        } else if (lastWeek.getTime() < messageDate.getTime()) {
+            let days = [qsTr("Sunday"),
+                        qsTr("Monday"),
+                        qsTr("Tuesday"),
+                        qsTr("Wednesday"),
+                        qsTr("Thursday"),
+                        qsTr("Friday"),
+                        qsTr("Saturday")];
+            return days[messageDate.getDay()];
+        } else {
+            return messageDate.getMonth()+1+"/"+messageDate.getDate()+"/"+messageDate.getFullYear()
```

Comment:
Could this date format change to use https://doc.qt.io/qt-5/qtquick-internationalization.html#6-internationalize-dates-times-and-currencies ?

Reply:
So last time I used the `toLocaleString()` API it didn't work. It was as if the API returns `undefined` in QML environment. But I can try again